### DISCUSSION TITLE
Warn about absolute URL in custom CSS/logo settings

### DISCUSSION THREAD

**Discussion by Kixunil** [btcpayserver/btcpayserver#4195] [Ideas / Feature Requests]
## Background

A customer entered absolute URL (with hostname) to CSS and logo settings. The URL was pointing to local storage. This broke when the URL was set to onion URL and the instance was later accessed through non-tor browser using clearnet domain (the server in question has both). Changing the URL to relative fixed it (by removing `https://domain`).

The solution is not obvious to non-technical users.

## Proposed improvement

When an URL is entered containing same domain as the domain of BTCPayServer (currently opened) a warning is shown saying: "Your URL is absolute, which may cause unexpected issues." and a button "Automatically change to relative" - after clicking on it the domain is stripped from the URL and the warning disappears.

## Alternative solution

Always strip the domain without asking the user when storing the settings. This would make it more fool-proof but may be surprising or scary to less knowledgeable users.

## A more fancy version

One of the above plus the root path is also replaced with some placeholder variable that's substituted. If the root path changes at any time the URLs will keep working correctly without the user having to change them. Note that root path placeholder should be inserted (at the beginning) even when it's not set - for the case it will be set in the future.

**Comment by bolatovumar** [comment]
PR implementing this here: https://github.com/btcpayserver/btcpayserver/pull/4354
